resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -61,10 +61,8 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private val runningExecutorsToPods = new mutable.HashMap[String, Pod]
   // Indexed by executor pod names and guarded by RUNNING_EXECUTOR_PODS_LOCK.
   private val runningPodsToExecutors = new mutable.HashMap[String, String]
-  // TODO(varun): Get rid of this lock object by my making the underlying map a concurrent hash map.
-  private val EXECUTOR_PODS_BY_IPS_LOCK = new Object
-  // Indexed by executor IP addrs and guarded by EXECUTOR_PODS_BY_IPS_LOCK
-  private val executorPodsByIPs = new mutable.HashMap[String, Pod]
+  private val executorPodsByIPs: concurrent.Map[String, Pod] = new
```

Comment:
@erikerlandson thank you for an explanation. That makes sense for me, though when I typed the line in scala console(2.11.8), the compiler was able to inference concurrent.Map[K, V] class. 

```
scala> val executorPodsByIPs = new ConcurrentHashMap[String, String]().asScala
executorPodsByIPs: scala.collection.concurrent.Map[String,String] = Map()
```

I looked into the docs, and found that .asScala method going to do a conversion from java.util.ConcurrentHashMap[K, V] to scala.concurrent.Map[K, V]

[mapAsScalaConcurrentMapConverter](http://www.scala-lang.org/api/2.11.8/index.html#scala.collection.JavaConverters$@mapAsScalaConcurrentMapConverter[A,B](m:java.util.concurrent.ConcurrentMap[A,B]):scala.collection.convert.Decorators.AsScala[scala.collection.concurrent.Map[A,B]])